homeassistant/components/sma_manager/sensor.py

```diff
@@ -0,0 +1,356 @@
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfPower, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+# Local Imports
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Create and sets up the entities and setup Riemann Integral version of the same entities, so they can be used in the energy panel.
+
+    @param hass:
+    @param config_entry:
+    @param async_add_entities:
+    @return:
+    """
+
+    sma = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = Coordinator(hass, sma)
+    device = DeviceInfo(
+        identifiers={(DOMAIN, sma.serial_number)},
+        name=sma.name,
+        manufacturer="SMA",
+        model="SMA Manager",
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+
+    # Create base entities
+    base_entities = [
+        GridConsumption(coordinator, sma, device),
+        GridFeed(coordinator, sma, device),
+        PhaseOneConsumption(coordinator, sma, device),
+        PhaseOneFeed(coordinator, sma, device),
+        PhaseTwoConsumption(coordinator, sma, device),
+        PhaseTwoFeed(coordinator, sma, device),
+        PhaseThreeConsumption(coordinator, sma, device),
+        PhaseThreeFeed(coordinator, sma, device),
+    ]
+    async_add_entities(base_entities)
+
+    # Create Integral Sensor for every base entity
+    integral_entities = []
+    for entity in base_entities:
+        name = f"{entity.name} Energy"
+        integral_entities.append(
+            IntegrationSensor(
+                integration_method=METHOD_TRAPEZOIDAL,
+                name=name,
+                round_digits=2,
+                source_entity=f"sensor.{entity.unique_id}",
+                unique_id=name.lower().replace(" ", "_"),
+                unit_prefix="k",
+                unit_time=UnitOfTime.HOURS,
+            )
+        )
+    async_add_entities(integral_entities)
+
+
+class Coordinator(DataUpdateCoordinator):
+    """Coordinates data fetching from the API."""
+
+    def __init__(self, hass: HomeAssistant, sma: SMA) -> None:
+        """Init the coordinator object.
+
+        @param hass:
+        @param sma:
+        """
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="SMA Coordinator",
+            update_interval=timedelta(seconds=sma.refresh_time),
+        )
+        self.sma = sma
+
+    async def _async_update_data(self):
+        """Refresh data from SMA Manager."""
+        await self.sma.refresh_data()
+
+
+class SensorBase(CoordinatorEntity, SensorEntity):
```

Comment:
Please read up on entity naming requirements and do the required asjustments: https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations